sys/net/network_layer/ng_sixlowpan/iphc/ng_sixlowpan_iphc.c

```diff
@@ -89,23 +89,24 @@ static inline bool _context_overlaps_iid(ng_sixlowpan_ctx_t *ctx,
              (iid->uint8[(ctx->prefix_len / 8) - 8] & byte_mask[ctx->prefix_len % 8])));
 }
 
-bool ng_sixlowpan_iphc_decode(ng_pktsnip_t *pkt)
+size_t ng_sixlowpan_iphc_decode(ng_pktsnip_t *ipv6, ng_pktsnip_t *pkt, size_t offset)
 {
     ng_netif_hdr_t *netif_hdr = pkt->next->data;
     ng_ipv6_hdr_t *ipv6_hdr;
     uint8_t *iphc_hdr = pkt->data;
-    uint16_t payload_offset = NG_SIXLOWPAN_IPHC_HDR_LEN;
+    size_t payload_offset = NG_SIXLOWPAN_IPHC_HDR_LEN;
     ng_sixlowpan_ctx_t *ctx = NULL;
-    ng_pktsnip_t *payload;
-    ng_pktsnip_t *ipv6 = ng_pktbuf_add(NULL, NULL, sizeof(ng_ipv6_hdr_t),
-                                       NG_NETTYPE_IPV6);
 
-    if (ipv6 == NULL) {
-        DEBUG("6lo iphc: error allocating ipv6 header space\n");
-        return false;
-    }
+    DEBUG("6lo iphc: ng_sixlowpan_iphc_decode(%p, %p, %u)\n",
+          (void *)ipv6, (void *)pkt, (unsigned)offset);
+
+    assert(ipv6 != NULL);
+    assert(ipv6->size >= sizeof(ng_ipv6_hdr_t));
 
     ipv6_hdr = ipv6->data;
+    iphc_hdr += offset;
+
+    od_hex_dump(iphc_hdr, pkt->size - offset, 0);
```

Comment:
Well..yeah but I had to remove it because it was complaining about undeclared(or undefined, don't remember) function.